group-size-refinement-list/react-instantsearch/src/lib/GroupSizeRefinementList.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from "react";
+import { Highlight, connectRefinementList } from "react-instantsearch-dom";
+
+import "./GroupSizeRefinementList.css";
+
+class GroupSizeRefinementList extends Component {
+    constructor(props) {
+        super(props);
+        const showMore = this.props.translations && this.props.translations.showMore ?
+            this.props.translations.showMore :
+            (expanded) => expanded ? 'Show less' : 'Show more';
+
+        this.state = {
+            showMore,
+            sortGroupByNbResults: props.sortGroupByNbResults !== undefined ? props.sortGroupByNbResults : true,
+            sortSizesByNbResults: props.sortSizesByNbResults !== undefined ? props.sortSizesByNbResults : true,
+            expanded: !props.showMore,
+            nbGroups: props.nbGroups ? props.nbGroups : 1
+        };
+    }
+
+    render() {
+        const { expanded, showMore } = this.state;
+
+        let inclusionArrays = [];
```

Comment:
A couple of points:

1. You can gain in clarity if you split this up.
For example have a method called `getSizeGroups`. It's important to keep render methods easy to read, as they handle all the rendering of the widget.

2. I would try to avoid recomputing the groups at every render. You could use memoization or precompute the value in a state.

Reply:
Can you please detail a bit more the second point? What do you have in mind ? I compute an array of array of sizes in the state ? Is this really more optimised compare to a regex match ?

Reply:
Can you please detail a bit more the second point? What do you have in mind ? I compute an array of array of sizes in the state ? Is this really more optimised compare to a regex match ?

Reply:
It's not exactly a regexp match only this code is doing. it's also manipulating collections. It's doing all these operations:
- reduce
- sort
- map
- filter

My point is theses operations happen on every render, often yielding the same results over and over because nothing change on the component. If you have many instances of this refinement list component, you might experience slowness. 
If you compute sizeGroups every time the relevant props have changed and store it in the state, you render method will be lighter, more readable and more performant. For example in `static getDerivedStateFromProps(props, state)`

Reply:
I started to write a new version but realised that I need the updated count to order sizeGroups by nbResults and Sizes inside a group per nbResults..